cmake/compiler/arcmwdt/compiler_flags.cmake

```diff
@@ -168,6 +168,8 @@ set_compiler_property(PROPERTY imacros -imacros)
 # Security canaries.
 #no support of -mstack-protector-guard=global"
 set_compiler_property(PROPERTY security_canaries -fstack-protector-all)
+set_compiler_property(PROPERTY security_canaries_strong -fstack-protector-strong)
+set_compiler_property(PROPERTY security_canaries_explicit -fstack-protector-explicit)
 
```

Comment:
`-fstack-protector-all` and `-fstack-protector` are different options. User should be able to select them different. If no explicit choice is specified by user, `-fstack-protector-all` can be taken to make it compatibility with previous versions